src/common/state/redis.c

```diff
@@ -356,15 +368,15 @@ void redis_object_table_get_entry(redisAsyncContext *c,
   REDIS_CALLBACK_HEADER(db, callback_data, r)
   redisReply *reply = r;
 
-  int *managers = malloc(reply->elements * sizeof(int));
+  db_client_id *managers = malloc(reply->elements * sizeof(db_client_id));
   int64_t manager_count = reply->elements;
 
   if (reply->type == REDIS_REPLY_ARRAY) {
     const char **manager_vector = malloc(manager_count * sizeof(char *));
-    for (int j = 0; j < reply->elements; j++) {
+    for (int j = 0; j < reply->elements; ++j) {
       CHECK(reply->element[j]->type == REDIS_REPLY_STRING);
-      managers[j] = atoi(reply->element[j]->str);
-      redis_get_cached_service(db, managers[j], manager_vector + j);
+      memcpy(managers[j].id, reply->element[j]->str, sizeof(db_client_id));
```

Comment:
Oh, that's because you were assigning to `id` instead of the whole struct.
Just assign to the whole struct.